ipaplatform/debian/tasks.py

```diff
@@ -88,4 +98,99 @@ def configure_pkcs11_modules(self, fstore):
     def restore_pkcs11_modules(self, fstore):
         pass
 
+    def insert_ca_certs_into_systemwide_ca_store(self, ca_certs):
+        result = True
+
+        # pylint: disable=ipa-forbidden-import
+        from ipalib import x509  # FixMe: break import cycle
+        from ipalib.errors import CertificateError
+        # pylint: enable=ipa-forbidden-import
+
+        # TODO: it would be nice to have paths.IPA_P11_KIT created here, so
+        # that it could still be used by p11-kit in the future if Debian were
+        # to adopt Red Hat's CA certificate management code. If we were to do
+        # that, the code should be split out into a separate method that both
+        # DebianTaskNamespace and RedHatTaskNamespace can call.
+
+        ca_certificates_dir_path = Path(paths.CA_CERTIFICATES_DIR)
+        try:
+            ca_certificates_dir_path.mkdir(mode=0o755, exist_ok=True)
+        except OSError as e:
+            logger.error(
+                'Could not create %s: %s', ca_certificates_dir_path, e)
+            return False
+
+        for i, ca_cert in enumerate(ca_certs):
+            cert, nickname, trusted, _ext_key_usage = *ca_cert,
+
+            if not trusted:
+                continue
+
+            try:
+                # XXX is rfc4514_string really displaying the DN
+                # elements in the wrong order?
```

Comment:
LDAP and X.509 have different opinions how to display a DN. LDAP (RFC 4514) uses leaf to root order, X.509 root to leaf order.

Reply:
Right... too much staring at the output of gnutls vs openssl x509 vs ldapsearch all day long... :)

Reply:
Hold on... if I load a certificate with `cryptography` and look at `cert.subject.rdns` then I get:

```
[<RelativeDistinguishedName(O=IPADOMAIN)>,
 <RelativeDistinguishedName(CN=Certificate Authority)>]
```

So I'd expect `rfc_4514_string` to show them in the opposite order--but it doesn't, it's outputting `O=IPADOMAIN,CN=Certificate Authority`!

Reply:
My understanding is that `cryptography` doesn't really implement explicit ordering. It simply traverses all attributes in the list of attributes for the name as they are. And that list is constructed by using the same order as it was passed on the name instantiation. See https://cryptography.io/en/latest/_modules/cryptography/x509/name/#Name.rfc4514_string and around.

Reply:
I'm getting the impression that I should avoid using module in future! :)